site/docs/features/gc-internals.md

```diff
@@ -0,0 +1,182 @@
+and to purge orphan files.
+
+Modules that supplement the `gc-base` module:
+
+* `gc-iceberg` implements the Iceberg table-format specific functionality.
+* `gc-iceberg-files` implements file listing + deletion using Iceberg's `FileIO`.
+* `gc-iceberg-mock` is a testing-only module to generate mock metadata, manifest-lists, manifests
+  and (empty) data files.
+* `gc-repository-jdbc` implements the live-content-sets-store using JDBC (PostgreSQL and compatible)
+  .
+* `s3mock` is a testing-only module containing a S3 mock backend that allows listing objects and
+  getting objects programmatically.
+* `s3mino` is a junit 5 test extension providing a Minio based S3 backend.
+
+The `gc-tool` module is a command-line interface, a standalone tool provided as an executable,
+it is an uber jar prefixed with a shell script, and can still be executed with `java -jar ...`.
+
+## Basic Nessie-GC functionality
+
+Nessie-GC implements a mark-and-sweep approach, a two-phase process:
+
+The "mark phase", or "live content identification", walks all named references and collects
+references to all `Content` objects that are considered as live. Those references are stored in a
+repository as a "live contents set". The "mark phase" is implemented in `IdentifyLiveContents`.
+
+The "sweep phase", or "delete orphan files", operates per content-id. For each content, all live
+versions of a `Content` are scanned to identify the set of live data files. After that, the
+base-location(s)  are scanned and all files that are not in the set of live data files are deleted.
+The "sweep phase" is implemented by `DefaultLocalExpire`.
+
+## Inner workings
+
+To minimize the amount of data needed to match against the set of live data files for a `Content`,
+the implementation does not actually remember all individual data files, like maintaining a
+`java.util.Set` of all those data files, but remembers all data files in a bloom filter.
+
+Both the "mark" (identify live contents) and "sweep" (identify and delete expired contents) phases
+provide a configurable _parallelism_: the number of concurrently scanned named references can be
+configured and the amount of concurrently processed tables can be configured.
+
+### _Mark_ phase optimization
+
+The implementation that walks the commit logs can be configured with a `VisitedDeduplicator`, which
+is meant to reduce the work required during the "mark" phase, if the commit to be examined has
+already been processed.
+
+There is a `DefaultVisitedDeduplicator` implementation, but it is likely that it requires too much
+memory during runtime, especially when the identify-run is configured with multiple GC policies
+and/or has to walk many commits. This `DefaultVisitedDeduplicator` is present, but due to the
+mentioned concerns _not_ available in the Nessie GC tool and the use of
+`DefaultVisitedDeduplicator` is not supported at all, and not recommended.
```

Comment:
This (and similar) info is, IMHO, to much low-level for a projectnessie.org docs page... I suppose it might confuse new users. I believe it might be preferable to keep only high level info on the project's side and move this into a `.md` file under the java module so that people more involved with the actual code can read it... we can, of course, add a link form site docs to the source tree.... but I do not insist :)